docs/PIPELINES_CONFIGURATION.md

```diff
@@ -0,0 +1,126 @@
+
+1. Install the [Openshift Pipelines Operator](https://docs.redhat.com/en/documentation/openshift_container_platform/4.6/html/pipelines/installing-pipelines#installing-pipelines).
+
+2. Ensure that the `pipeline-as-code-controller` is up by getting its route:
+
+```
+kubectl get route -n <the-operator's-namespace> pipelines-as-code-controller
+```
+
+3. Download `cosign`, which will be used to generate the updated `signing-secrets`.
+
+```
+curl -sL https://github.com/sigstore/cosign/releases/latest/download/cosign-linux-amd64 -o /usr/bin/cosign && chmod +x /usr/bin/cosign
+```
+
+4. In your Operator's Namespace, delete (if exists) the `signing-secrets` Secret.
+
+5. Generate the new `signing-secrets` inside the Operator's Namespace and patch the new secret as immutable:
+
+```
+export KUBERNETES_SERVICE_PORT=<your-kubernetes-service port>
+export KUBERNETES_SERVICE_HOST=<your-kubernetes-service host>"
+cosign generate-key-pair k8s://<your-operator's namespace>/signing-secrets
+kubectl patch secret -n <your-operator's namespace> signing-secrets -o yaml --patch='{"immutable": true}'
+```
+
+6. Ensure that the `tektonconfigs` CRDs are available. You can verify that if the below command returns 1 as response:
+
+```
+kubectl api-resources | grep -c "^tektonconfigs "
+```
+
+7. Update the `TektonConfig`, by enabling the necessary resolvers and use a targeted namespace:
+
+```
+kubectl patch tektonconfig config --type 'merge' --patch "$( cat <<EOF
+spec:
+  pipeline:
+    enable-bundles-resolver: true
+    enable-cluster-resolver: true
+    enable-custom-tasks: true
+    enable-git-resolver: true
+    enable-hub-resolver: true
+    enable-tekton-oci-bundles: true
+  chain:
+    artifacts.oci.storage: oci
+    artifacts.pipelinerun.format: in-toto
+    artifacts.pipelinerun.storage: oci
+    artifacts.taskrun.format: in-toto
+    artifacts.taskrun.storage: oci
+    transparency.enabled: "true"
+    transparency.url: http://rekor-server.\$NAMESPACE.svc
+EOF
+)"
+```
+
+8. Create the pipelines secret, containing your Github App's `Webhook URL` and `Webhook Secret`.
+
+```
+export APP_NAMESPACE=<your-app's namespace>
+export PIPELINES_SECRET_NAME=<your pipelines-secret name>
+export GITHUB_APP_WEBHOOK_SECRET=<your github app's webhook secret>
+export GITHUB_APP_WEBHOOK_URL=<your github app's webhook url>
+kubectl -n "$APP_NAMESPACE" create secret generic "$PIPELINES_SECRET_NAME" \
+    --from-literal="webhook-github-secret=$GITHUB_APP_WEBHOOK_SECRET" \
+    --from-literal="webhook-url=$GITHUB_APP_WEBHOOK_URL"
+```
+
+9. Create the `pipelines-as-code-secret`, containing your Github App's `App ID`, `Private Key`, `Webhook Secret`.
+
+```
+kubectl -n "$PIPELINES_NAMESPACE" create secret generic pipelines-as-code-secret \
+    --from-literal github-application-id="$GITHUB_APP_APP_ID" \
+    --from-literal github-private-key="$GITHUB_APP_PRIVATE_KEY" \
```

Comment:
A guide on how to properly set the `GITHUB_APP_PRIVATE_KEY` would be handy. The first time I passed it in, I didn't format it correctly and got an error from PaC

Reply:
I've added more info here: [4501a65](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/19/commits/4501a6546c4814f45b6743eb93dbca1e5c9fcdd3)